adapter.go

```diff
@@ -59,18 +60,30 @@ func finalizer(a *adapter) {
 
 // NewAdapter is the constructor for Adapter. If database name is not provided
 // in the Mongo URL, 'casbin' will be used as database name.
-func NewAdapter(url string, timeout ...interface{}) (persist.Adapter, error) {
-	if !strings.HasPrefix(url, "mongodb+srv://") && !strings.HasPrefix(url, "mongodb://") {
-		url = fmt.Sprint("mongodb://" + url)
+func NewAdapter(uri string, timeout ...interface{}) (persist.Adapter, error) {
+	if !strings.HasPrefix(uri, "mongodb+srv://") && !strings.HasPrefix(uri, "mongodb://") {
+		uri = fmt.Sprint("mongodb://" + uri)
 	}
-	clientOption := options.Client().ApplyURI(url)
+	clientOption := options.Client().ApplyURI(uri)
 
-	return NewAdapterWithClientOption(clientOption, timeout...)
+	u, err := url.Parse(uri)
+	if err != nil {
+		return nil, err
+	}
+
+	var databaseName string
+	if u.Path != "" {
+		databaseName = u.Path[1:]
+	} else {
+		databaseName = "casbin"
```

Comment:
casbin_rule

```go
	// Or you can use an existing DB "abc" like this:
	// The adapter will use the table named "casbin_rule".
	// If it doesn't exist, the adapter will create it automatically.
	// a := mongodbadapter.NewAdapter("127.0.0.1:27017/abc")
```

Reply:
sry, really ashamed of the low-level mistake I made